ipaserver/install/dns.py

```diff
@@ -165,17 +165,19 @@ def _setup_dns_over_tls(options):
             f.write("\n".join(dns_none))
         nm.reload_or_restart()
 
-    # Overwrite resolv.conf to point to IPA
+    # Ensure resolv.conf points to IPA
     cfg = [
         "# auto-generated by IPA installer",
         "search .",
         "nameserver 127.0.0.1\n"
     ]
-    fstore = sysrestore.FileStore(paths.SYSRESTORE)
-    fstore.backup_file(paths.RESOLV_CONF)
-    with open(paths.RESOLV_CONF, 'w') as f:
-        f.write('\n'.join(cfg))
-        os.chmod(paths.RESOLV_CONF, 0o644)
+    nameservers = dnsforwarders.get_dnspython_nameservers()
+    if not nameservers or nameservers[0] != "127.0.0.1":
```

Comment:
dnsforwarders.get_dnspython_nameservers() returns a list of IPv4Address and IPv6Address objects. The comparison to a string will not work.
You could use get_ipa_resolver().nameservers (from ipapython.dnsutil) this is used in get_dnspython_nameservers: https://github.com/freeipa/freeipa/blob/master/ipalib/install/dnsforwarders.py#L99